core/src/main/java/io/ultrabrew/metrics/util/DistributionBucket.java

```diff
@@ -0,0 +1,66 @@
+package io.ultrabrew.metrics.util;
+
+public class DistributionBucket {
+
+    private static final String UNDERFLOW = "underflow";
+    private static final String OVERFLOW = "overflow";
+
+    private int[] buckets;
+
+    public DistributionBucket(final int[] buckets) {
+        this.buckets = buckets;
+    }
+
+    public int getCount() {
+        return buckets.length + 1; // includes the underflow and overflow buckets
+    }
+
+    public int getBucketIndex(int latency) {
+        return binarySearch(latency);
+    }
+
+    public String[] getBucketNames() {
+        int bucketCount = buckets.length;
+        String[] names = new String[bucketCount + 1];
+        int i = 0;
+        names[i] = UNDERFLOW;
```

Comment:
tracking underflow should be based on user preference to avoid wasting a bucket for uses where it's not relevant. minor optimization.